### PR TITLE
EV charger energy management changes.

### PR DESCRIPTION
**Pull request by rstraczynski**

### CODE REVIEW

File: test/suite/helper.go [hunk 1]
```diff
@@ -114,3 +114,73 @@ func FloatMapMessage(topic, messageType, service string, value map[string]float6
 		),
 	}
 }
+
```
Comment (boguslaw-wojcik) [inline test/suite/helper.go:117]: If you're making builder, make one, and add a method Bool/String/Etc in it.

Comment (rstraczynski) [reply test/suite/helper.go:117]: Good idea!

File: adapter/service/chargepoint/service.go [hunk 2]
```diff
@@ -287,95 +271,24 @@ func (s *service) SupportedStates() []string {
...
-	mode, err := controller.ChargepointChargingModeReport()
-	if err != nil {
-		return false, fmt.Errorf("%s: failed to get charging mode report: %w", s.Name(), err)
-	}
-
-	if !force && !s.reportingCache.ReportRequired(s.stateReportingStrategy, EvtChargingModeReport, "", mode) {
-		return false, nil
+// normalizeChargingMode normalizes provided charging mode. Returns true, when everything is fine.
+func (s *service) normalizeChargingMode(mode string) (string, error) {
+	if mode == "" {
```
Comment (boguslaw-wojcik) [inline adapter/service/chargepoint/service.go:276]: I wonder if we should send normal here if empty...

File: test/suite/helper.go (hunk 1 above)
Comment (rstraczynski) [reply test/suite/helper.go:117]: Fixed.

File: adapter/service/chargepoint/service.go (hunk 2 above)
Comment (rstraczynski) [reply adapter/service/chargepoint/service.go:278]: As we discussed...

**Review by boguslaw-wojcik — APPROVED** [review]

**Pull request merged**